### PR TITLE
Add taggings UI and API

### PR DESCRIPTION
**Pull request by jrp6**
DO NOT MERGE before Aalto-LeTech/django-colortag#2

### CODE REVIEW

File: course/api/views.py [hunk 1]
```diff
@@ -98,20 +98,29 @@ class CourseUsertagsViewSet(NestedViewSetMixin,
...
                  .select_related('tag', 'user', 'user__user')
-                 .only('tag__id', 'tag__course_instance',
+                 .only('tag__id', 'tag__course_instance', 'tag__name', 'tag__slug',
                        'user__user__id', 'user__user__username', 'user__student_id',
                        'course_instance__id')
                  .all() )
     parent_lookup_map = {'course_id': 'course_instance_id'}
 
+    def get_serializer_context(self):
+        context = super(CourseUsertaggingsViewSet, self).get_serializer_context()
```
Comment (raphendyr) [inline course/api/views.py]: `super()` is good enough. aplus doesn't support python2.7 anymore

**Comment by raphendyr** [comment]
seems good!

**Comment by jrp6** [comment]
Fixed `super()` and switched requirement to Aalto-LeTech/django-colortag@1.3.0

File: course/api/views.py [hunk 2]
```diff
@@ -98,20 +98,29 @@ class CourseUsertagsViewSet(NestedViewSetMixin,
...
                                 CourseModuleResourceMixin,
                                 CourseResourceMixin,
-                                viewsets.ReadOnlyModelViewSet):
+                                mixins.CreateModelMixin,
+                                mixins.RetrieveModelMixin,
+                                mixins.DestroyModelMixin,
+                                mixins.ListModelMixin,
+                                viewsets.GenericViewSet):
     permission_classes = api_settings.DEFAULT_PERMISSION_CLASSES + [
         OnlyCourseTeacherPermission,
```
Comment (raphendyr) [inline course/api/views.py]: I need to check that GraderUser is not allowed to access this endpoint.

**Pull request merged**